godot-core/src/builtin/vectors/vector_macros.rs

```diff
@@ -251,233 +251,419 @@ macro_rules! impl_vector_index {
-                        b.$comp, pre_a.$comp, post_b.$comp, weight, b_t, pre_a_t, post_b_t,
-                    );
-                )*
+                Self::new(
+                    $(
+                        self.$comp.cubic_interpolate_in_time(
+                            b.$comp, pre_a.$comp, post_b.$comp, weight, b_t, pre_a_t, post_b_t,
+                        )
+                    ),*
+                )
+            }
+
+            /// Returns the normalized vector pointing from this vector to `to`. This is equivalent to using `(b - a).normalized()`.
+            #[inline]
+            pub fn direction_to(self, to: Self) -> Self {
+                (to - self).normalized()
+            }
+
+            /// Returns the squared distance between this vector and `to`.
+            ///
+            /// This method runs faster than [`Self::distance_to`], so prefer it if you need to compare vectors or need the squared distance for some formula.
+            #[inline]
+            pub fn distance_squared_to(self, to: Self) -> real {
+                (to - self).length_squared()
+            }
+
+            /// Returns the distance between this vector and `to`.
+            #[inline]
+            pub fn distance_to(self, to: Self) -> real {
+                (to - self).length()
+            }
 
-                Self::new($($comp),*)
+            /// Returns the dot product of this vector and `with`.
+            #[inline]
+            pub fn dot(self, with: Self) -> real {
+                self.to_glam().dot(with.to_glam())
             }
 
+            /// Returns a new vector with all components rounded down (towards negative infinity).
+            #[inline]
+            pub fn floor(self) -> Self {
+                Self::from_glam(self.to_glam().floor())
+            }
+
+            /// Returns true if each component of this vector is finite.
+            #[inline]
+            pub fn is_finite(self) -> bool {
+                self.to_glam().is_finite()
+            }
+
+            /// Returns `true` if the vector is normalized, i.e. its length is approximately equal to 1.
+            #[inline]
+            pub fn is_normalized(self) -> bool {
+                self.to_glam().is_normalized()
+            }
+
+            /// Returns `true` if this vector's values are approximately zero.
+            /// This method is faster than using [`Self::is_equal_approx`] with one value as a zero vector.
+            #[inline]
             pub fn is_zero_approx(self) -> bool {
-                $(self.$comp.is_zero_approx())&&*
+                $( self.$comp.is_zero_approx() )&&*
             }
 
-            pub fn posmod(self, pmod: $Scalar) -> Self {
+            /// Returns the result of the linear interpolation between this vector and `to` by amount `weight`.
+            /// `weight` is on the range of `0.0` to `1.0`, representing the amount of interpolation.
+            #[inline]
```

Comment:
For the docs, you should have an empty line between the short description and further elaboration. It doesn't matter that much for struct methods, but if there's ever an overview page (like in modules), then only the first paragraph is displayed, and it should be as brief as possible.

Also, `Self::is_equal_approx` link doesn't work, see [generated docs](https://godot-rust.github.io/docs/gdext/pr-721/godot/builtin/struct.Vector3.html#method.is_zero_approx).

Reply:
The missing a the empty line is an error, thanks for catching that. I looked at `is_zero_approx()` again and [Godot's docs](https://docs.godotengine.org/de/4.x/classes/class_vector3.html#class-vector3-method-is-zero-approx) has a line break there.